src/mol-plugin-state/transforms/data.ts

```diff
@@ -134,6 +137,28 @@ const DownloadBlob = PluginStateTransform.BuiltIn({
     // }
 });
 
+type DeflateData = typeof DeflateData
+const DeflateData = PluginStateTransform.BuiltIn({
+    name: 'defalate-data',
+    display: { name: 'Deflate', description: 'Deflate compressed data' },
+    params: {
+        method: PD.Select('gzip', [['gzip', 'gzip']]), // later on we might have to add say brotli
+        isString: PD.Boolean(false),
+        stringEncoding: PD.Optional(PD.Select('utf-8', [['utf-8', 'UTF8']])),
+        label: PD.Optional(PD.Text(''))
+    },
+    from: [SO.Data.Binary],
+    to: [SO.Data.Binary, SO.Data.String]
+})({
+    apply({ a }, plugin: PluginContext) {
+        return Task.create('Gzip', async ctx => {
+            const decompressedData = await ungzip(ctx, a.data);
+            // handle decoding based on stringEncoding param
+            return new SO.Data.Binary(decompressedData as Uint8Array, { label: a.label });
```

Comment:
Need to actually implement the bytes -> UTF8 and return `SO.Data.String` if the `isString` param is true.

You can use the `utf8Read` function in `mol-io/common/utf8.ts`.